azavea/cac-tripplanner-android
# Upgrade android SDK version to 30

## Overview

This PR:
- updates `targetSdkVersion` and `compileSdkVersion` to version 30
- updates `minSdkVersion` to version 21

### Notes

The update of `minSdkVersion` to version 21 has two reasons:
1. Tested the app before upgrading the min version and running on the virtual device with sdk version 19 failed. The reason seemed to be two-folds:
    - ` Unable to get provider com.google.firebase.provider.FirebaseInitProvider` error occurred. Some google search indicated that adding `Multidex` support would fix the issue, which then opened up a can of warms.
    - `okHttp` reported an error `Expected Android API level 21+ but was 19`. The suggestion here was obvious, plus starting from sdk version 21, `Multidex` support is built-in, which means no code base change needed
2. This chart (Launch Android Studio -> File -> New -> New project -> Next). It should be fine if we do such an upgrade.

![Screen Shot 2022-07-14 at 5 05 53 PM](https://user-images.githubusercontent.com/16109558/179087996-6881c354-9fec-4ee4-99a2-395cd4805ab2.png)

## Testing Instructions

- Follow instructions from README to set up the project
   - Note that both `api_keys.xml` and `google-services.json` can be found in LastPass
   - For the Google Maps API key:
       - Make sure to follow the URL in the comment section of `api_keys.xml`, log in with your Google Account, and use the debug token rather than the release token. And paste this debug token in your local `api_keys.xml`
       - If your fingerprint cert is not added under that token, make sure to append yours to it in the Google Cloud console. If your Google account does not have access, please ping @BryanQuigley to assign you permission.
- Open the app in an emulator with virtual devices on SDK version 21 and version 30. Make sure the app functions well on both. Click around in the app and make sure all feature work fine as expected.

Closes https://github.com/azavea/cac-tripplanner-android/issues/196
